File: prob12.js
// https://projecteuler.net/problem=12

let currentNum = 1;
let nextNum = 2;
let largestDivCount = 0;
let largestNumber = 0;

while(largestDivCount < 500){
  let divCount = 0;
  let finalDivCount;
  let latestDiv;

  for(let i = 1; i <= Math.round(Math.sqrt(currentNum)); i++){
    if(currentNum % i === 0){
      latestDiv = i;
      divCount++;
    }
  }

  if(Math.pow(latestDiv, 2) === currentNum){
    finalDivCount = divCount * 2 - 1;
  }else{
    finalDivCount = divCount * 2;
  }

  if(finalDivCount > largestDivCount){
    largestDivCount = finalDivCount;
    largestNumber = currentNum;
  }

  currentNum += nextNum;
  nextNum++;
}

console.log(largestNumber);

File: prob12.py
# https://projecteuler.net/problem=12
# Completed 1/2/2022

import math

def factorCount(num):
  factors = 0
  for i in range(1, math.ceil(math.sqrt(num))):
    if num % i == 0:
      factors += 2
  return factors

currentTriNum = 1
count = 1
found = False

while found == False:
  if factorCount(currentTriNum) > 500:
    print(currentTriNum)
    found = True
    break
  count += 1
  currentTriNum += count